src/stream/src/executor/mview/materialize.rs

```diff
@@ -643,8 +738,31 @@ impl<SD: ValueRowSerde> MaterializeCache<SD> {
                                 true
                             };
                             if need_overwrite {
-                                change_buffer.update(key.clone(), old_row.row.clone(), row.clone());
-                                self.lru_cache.put(key.clone(), Some(CompactedRow { row }));
+                                let final_row = if let Some(toastable_indices) =
+                                    toastable_column_indices
+                                {
+                                    // For TOAST-able columns, replace Debezium's unavailable value placeholder with old row values.
+                                    let old_row_deserialized =
```

Comment:
whats the behavior when old value looking up fails?

Reply:
Old row must exist in hummock, materialized executor will first loop up it in cache and then loop up hummock if cache miss.